File: database-go/www/handlers/status.go
package handlers

import (
	"github.com/labstack/echo/v4"
	"go-kv-database/app"
	"go-kv-database/storage"
	"net/http"
)

func Status(c echo.Context) error {
	return c.JSON(http.StatusOK, map[string]any{
		"version":          app.Version,
		"nodeId":           app.NodeId,
		"servicePort":      app.ServicePort,
		"serviceUrl":       app.ServiceUrl,
		"serviceLogLevel":  app.ServiceLogLevel,
		"isWarmup":         app.IsWarmup,
		"meshNetworkUrl":   app.MeshNetworkUrl,
		"meshNodes":        app.MeshNodes,
		"maxKeyLength":     app.MaxKeyLength,
		"maxValueLength":   app.MaxValueLength,
		"availableRecords": len(storage.GetAll()),
	})
}

File: packages/database/src/http/handlers/status.ts
import { env } from '../../lib/env';
import { RequestHandler } from '../../net/http';
import { dbMaxKeyLength, dbMaxValueLength } from '../../storages/nkv-database';

// . Get server settings [immutable]
export const status: RequestHandler = async (db) => {
  // Response
  return {
    version: '0.0.1',
    nodeId: env.nodeId,
    servicePort: env.servicePort,
    maxKeyLength: dbMaxKeyLength,
    maxValueLength: dbMaxValueLength,
    availableRecords: db.size()
  };
};
